### PR TITLE
libgcrypt-tests.py: add distribution specific test cases

### PR DESCRIPTION
**Pull request by nasastry**
Add distribution specific test cases.
With the support of yaml, can run distribution and upstream test cases.

Signed-off-by: Nageswara R Sastry <rnsastry@linux.ibm.com>

### CODE REVIEW

**Comment by nasastry** [comment]
```
RHEL:
JOB ID     : ff3cf218421bcadd8aa33fb5e7d6510fa91077de
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2022-09-21T02.25-ff3cf21/job.log
 (1/2) libgcrypt-tests.py:libgcrypt.test;run-run_type-upstream-94e9:  PASS (178.22 s)
 (2/2) libgcrypt-tests.py:libgcrypt.test;run-run_type-distro-cd12:  PASS (183.44 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2022-09-21T02.25-ff3cf21/results.html
JOB TIME   : 369.64 s


SLES:
JOB ID     : 8ef3fce6944c90c91b85310f2f4cb1750e22e351
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2022-09-21T12.52-8ef3fce/job.log
 (1/2) libgcrypt-tests.py:libgcrypt.test;run-run_type-upstream-94e9:  PASS (176.90 s)
 (2/2) libgcrypt-tests.py:libgcrypt.test;run-run_type-distro-cd12:  PASS (141.65 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2022-09-21T12.52-8ef3fce/results.html
JOB TIME   : 337.02 s
```

File: security/libgcrypt-tests.py [hunk 1]
```diff
@@ -34,33 +34,49 @@ def setUp(self):
...
-        deps = ['gcc', 'make']
-        if detected_distro.name in ['rhel', 'SuSE', 'fedora', 'centos']:
-            deps.extend(["autoconf", "automake", "libgpg-error-devel",
-                         "libgpg-error", "transfig"])
-        else:
+        deps = ['gcc', 'make', "autoconf", "automake", "libgpg-error-devel",
+                "libgpg-error"]
+        if detected_distro.name in ['rhel', 'fedora', 'centos']:
+            deps.extend(["transfig"])
+        elif 'Ubuntu' in detected_distro.name:
```
Comment (Naresh-ibm) [inline security/libgcrypt-tests.py:41]: @nasastry what about sles here?
isn't supported for this feature? if not, we can add as unsupported

**Comment by Naresh-ibm** [comment]
@nasastry also, for .py changes debug.job(if using nrunner) or job.log (if not using nrunner) will be needed for reference.
so please attach it.

File: security/libgcrypt-tests.py (hunk 1 above)
Comment (nasastry) [reply security/libgcrypt-tests.py:41]: @Naresh-ibm Already attached SLES related output in the above.
RHEL & SLES are supported.
It's just that SLES doesn't require any specific package as pre-requisite.

**Review by Naresh-ibm — APPROVED** [review]

**Comment by Naresh-ibm** [comment]
LGTM

**Pull request merged**